ethereum_clients/Plugin.js

```diff
@@ -201,16 +212,20 @@ class Plugin extends EventEmitter {
     return this.process && this.process.stop()
   }
   // public json rpc
-  async rpc(method, params = []) {
+  async rpc(method, params = [], id, result) {
```

Comment:
this needs some explanation: what is the result param doing? and what is the reasoning to have external logic control the rpc id?

Reply:
the `result` param is a part of the jsonrpc 2.0 spec:
https://www.jsonrpc.org/specification
See section `5 Response object` > `result`

External logic should be able to pass in the rpc id to track incoming data for a response to the request.

i.e. in clef, i send `{jsonrpc: '2.0', id: 5, method: 'clef_chainId', params: []}` which returns the chain id, but i pass in id:5 so i can keep checking incoming data for the proper response.